docs/extensions/sandbox/register.md

```diff
@@ -0,0 +1,128 @@
+---
+description: Learn about the ways to register extensions in the sandbox
+contributors: Rijk van Zanten
+---
+
+# Registering Extensions in the Sandbox
+
+The way extensions are registered in the sandbox closely follows the syntax of the existing extension types, but there
+are some subtle yet crucial differences in how data is passed back and forth to the API.
+
+## Hooks
+
+Hooks are registered by exporting a default function that gets an object with the `filter` and `action` properties that
+can be used to register filters and actions respectively.
+
+```js
+import { log } from 'directus:api';
+
+export default ({ filter, action }) => {
+	filter('items.create', (payload) => {
+		log('Creating Item!');
+	});
+
+	action('items.create', (payload) => {
+		log('Item created!');
+	});
+};
+```
+
+The callback functions for the hooks get the payload that triggered the event as the only parameter.
+
+### TypeScript
+
+You can import the `SandboxHookRegisterContext` type from `directus:api` to type the registration context object:
+
+```ts
+/// <reference types="@directus/extensions/api.d.ts" />
+
+import { log } from 'directus:api';
+import type { SandboxHookRegisterContext } from 'directus:api';
+
+export default ({ filter, action }: SandboxHookRegisterContext) => {
+	filter('items.create', (payload) => {
+		log('Creating Item!');
+	});
+
+	action('items.create', (payload) => {
+		log('Item created!');
+	});
+};
+```
+
+## Endpoints
+
+Endpoints are registered by exporting a default function that gets a `router` object as it's only parameter:
+
+```js
+export default (router) => {};
+```
```

Comment:
Can we no longer freely decide the root path of your endpoints by using the following format?
```js
export default {
	id: 'test',
	handler: (router) => {
		router.get('/', () => {
			return {
				status: 200,
				body: 'Hello World',
			}
		});
	}
}
```
Currently a lot of extensions have the required `directus-extension-` prefix which resulting in unwieldy paths like `http://localhost:8055/directus-extension-secure-endpoint`

Reply:
AFAIK this is a restriction by design

Reply:
If so thats fine but let's explicitly document that decision then 😄

Reply:
Definitely! IIRC we had that in the previous version